### PR TITLE
Horizontal accordion77

### LINKED ISSUE #77: Horizontal accordion
Seems to use a background image/video, which gets replaced with content when an item is selected.
Should use fragments like tabs.

Initial view, with video/default content:
Screenshot 2023-06-14 at 9 14 31 AM

Selected view changes:
Screenshot 2023-06-14 at 9 14 40 AM

Screenshot 2023-06-14 at 9 15 17 AM

Screenshot 2023-06-14 at 9 15 25 AM

Screenshot 2023-06-14 at 9 15 33 AM

Original Page example: https://sgcreative.southernglazers.com/bloominbrands/p/6

### PR DESCRIPTION
**Pull request by atopper**
## Issue

Fix #77

## 🔗 Test URLs:

- Before: https://main--sgws-creative--hlxsites.hlx.page/drafts/top/vertselector
- After: https://horizontal-accordion77--sgws-creative--hlxsites.hlx.page/drafts/top/vertselector

Original: https://sgcreative.southernglazers.com/bloominbrands/p/6

### CODE REVIEW

**Review by badvision — APPROVED** [review]
Looks fine to me.  Why do we need to use top/left instead of margin: auto?  Is this related to the general requirement for relative page sizing?

File: blocks/chart/chart.js [hunk 1]
```diff
@@ -703,4 +708,16 @@ export default function decorate(block) {
       }
     }, 500);
   });
+  // Trigger this event type to draw the charts immediately.
```
Comment (atopper) [inline blocks/chart/chart.js:711]: @tmathern  - I put in a custom event (`drawChart`) handler.  The load in the vertical selector was already delayed, and the extra 500ms even had me wondering if something was wrong.

File: blocks/vertical-selector/vertical-selector.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+  if (wasAlreadyDisplayed || index === 0) {
+    // Show the first (default) child again.
+    viewer.querySelector('div.default-viewer').classList.add('vs-active');
+    closeButton.classList.remove('vs-active');
+  } else {
+    const showViewer = viewer.querySelector(`div.fragment-viewer-${index}`);
+    showViewer.classList.add('vs-active');
+    closeButton.classList.add('vs-active');
+    window.dispatchEvent(new Event('drawChart'));
```
Comment (atopper) [inline blocks/vertical-selector/vertical-selector.js:87]: ☝️  Dispatching event to draw charts here.

File: blocks/chart/chart.js (hunk 1 above)
Comment (tmathern) [reply blocks/chart/chart.js:711]: Ok!

File: blocks/vertical-selector/vertical-selector.css [hunk 3]
```diff
@@ -0,0 +1,125 @@
+.vertical-selector .default-viewer video {
+  width: 100%;
+}
+
+.vertical-selector .default-viewer button {
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  background-color: #000000b3;
```
Comment (tmathern) [inline blocks/vertical-selector/vertical-selector.css:10]: That's always the same, not themable?

File: blocks/vertical-selector/vertical-selector.css [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  fill: var(--static-color-black);
+  color: var(--static-color-black);
+}
+
+.vertical-selector .close-viewer span.icon-close:not(.vs-active) {
+  opacity: 0;
+  display: none;
+}
+
+.vertical-selector .close-viewer span.icon-close.vs-active {
```
Comment (tmathern) [inline blocks/vertical-selector/vertical-selector.css:70]: Eventually the icons need to scale too

File: blocks/vertical-selector/vertical-selector.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  const videoLink = cell.querySelector(':scope a');
+  if (!videoLink?.href?.endsWith('.mp4')) {
+    return false;
+  }
+  const videoP = videoLink?.closest('p');
+  if (!videoP) {
+    return false;
+  }
+  const posterP = videoP.previousElementSibling;
+  const posterImage = posterP?.querySelector('img');
```
Comment (tmathern) [inline blocks/vertical-selector/vertical-selector.js:29]: When authoring, make sure this image is a jpeg.

**Review by tmathern — APPROVED** [review]

**Comment by tmathern** [comment]
I closed https://github.com/hlxsites/sgws-creative/pull/109 without merging, since the changes are integrated here! 👍

**Review by allentsliu — APPROVED** [review]
LGTM!

**Comment by tmathern** [comment]
@atopper For the lighthouse score, I suggest you use an actual product page and include the accordion here. https://main--sgws-creative--hlxsites.hlx.page/bloominbrands/p/6 should have theming. Please add the accordion on the page and not the product fragment, as the accordion varies per page whereas the product fragment gets reused.

**Pull request merged**